### PR TITLE
Introduce Docs and Release cycle

### PR DESCRIPTION
**Pull request by Billcountry**
## Introduce Docs and Release cycle
- Docs are automatically built from python doc_strings and pushed to the PR in circle
- Package built automatically and published to pypi on merge to master

### CODE REVIEW

File: tests/test_entity.py [hunk 1]
```diff
@@ -23,14 +33,17 @@ def test_value_setting(self):
     def test_invalid_values(self):
         entity = TestEntity()
         # Test setting invalid values
-        self.assertRaises(InvalidValueError, setattr, entity, "prop1", 10)  # Text property setting int
-        self.assertRaises(InvalidValueError, setattr, entity, "prop2", "text")  # Int property setting Text
-        self.assertRaises(InvalidValueError, setattr, entity, "prop2", 10.3)  # Int property setting Float
+        message = "10 is not a valid value for property prop1 of type TextProperty"
+        self.assertRaisesWithMessage(InvalidValueError, message, setattr, entity, "prop1", 10)  # Text property setting int
```
Comment (georgemarrows) [inline tests/test_entity.py:37]: Minor: isn't this clearer?
```suggestion
        self.assertRaisesWithMessage(InvalidValueError, message, lambda: entity.prop1 = 10)  # Text property setting int
```

File: tests/test_propterty.py [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+        self.assertNotIn("integer_property", self.entity.__datastore_entity__)
+        self.assertInvalidValues("integer_property", "10", 10.3, {})
+
+    def test_text_property(self):
+        self.entity.text_property = "Some text"
+        self.assertEqual(self.entity.__datastore_entity__.get("text_property"), "Some text")
+        self.assertInvalidValues("text_property", 10, {}, [])
+
+    def test_datetime_property(self):
+        self.assertIsInstance(self.entity.auto_now_add_property, datetime)
```
Comment (georgemarrows) [inline tests/test_propterty.py:49]: Minor: and check that it's within a certain range of the time the test ran?

Comment (Billcountry) [reply tests/test_propterty.py:49]: Done, checked between now before running the get and after

File: tests/test_propterty.py [hunk 3]
```diff
@@ -46,8 +46,10 @@ def test_text_property(self):
         self.assertInvalidValues("text_property", 10, {}, [])
 
     def test_datetime_property(self):
-        self.assertIsInstance(self.entity.auto_now_add_property, datetime)
         now = datetime.now()
+        self.assertIsInstance(self.entity.auto_now_add_property, datetime)
+        # Test that the vale was just assigned on read
+        self.assertTrue(now < self.entity.auto_now_add_property < datetime.now())
```
Comment (georgemarrows) [inline tests/test_propterty.py]: 👍

**Pull request merged**